src/services/EcoBridge/OmniBridge/OmniBridge.ts

```diff
@@ -0,0 +1,763 @@
+      return
+    }
+
+    const { feeManagerAddress, foreignToHomeFee, homeToForeignFee, currentDay } = feesData
+
+    this._currentDay = currentDay
+
+    const isAddressReward = await checkRewardAddress(
+      feeManagerAddress,
+      this._account,
+      this._staticProviders[this._homeChainId]
+    )
+
+    const feeType = this._activeChainId === this._homeChainId ? homeToForeignFee : foreignToHomeFee
+
+    const toAmount = isAddressReward
+      ? parsedFromAmount
+      : await fetchToAmount(
+          this.bridgeId,
+          feeType,
+          { address, chainId, name, mediator: fromTokenMediator },
+          {
+            address: toToken.address,
+            chainId: toToken.chainId,
+            name: toToken.name,
+            mediator: toToken.mediator ?? '',
+          },
+          parsedFromAmount,
+          feeManagerAddress,
+          this._staticProviders[this._homeChainId]
+        )
+
+    const feeAmount = parsedFromAmount.sub(toAmount)
+
+    let fee = '0%'
+
+    if (feeAmount.gt(0)) {
+      fee = `${(
+        (Number(formatUnits(feeAmount.toString(), toToken.decimals)) /
+          Number(formatUnits(parsedFromAmount.toString(), decimals))) *
+        100
+      ).toFixed(2)}%`
+    }
+
+    this._tokensPair = {
+      fromToken: {
+        address,
+        chainId,
+        name,
+        mode: fromTokenMode,
+        mediator: fromTokenMediator,
+        decimals: decimals ?? 0,
+        amount: parsedFromAmount,
+        symbol,
+      },
+      toToken: {
+        address: toToken.address,
+        chainId: toToken.chainId,
+        name: toToken.name,
+        decimals: toToken.decimals,
+        mode: toToken.mode,
+        mediator: toToken.mediator,
+      },
+    }
+
+    let gas: undefined | string = undefined
+
+    try {
+      const GAS_COST = 260000
+
+      const gasPrice = await this._staticProviders[this._activeChainId]?.getGasPrice()
+
+      if (!gasPrice) throw new Error('Cannot get gas price')
+
+      const {
+        bundle: { nativeCurrencyPrice },
+      } = await request(subgraphClientsUris[this._activeChainId as SWPRSupportedChains], QUERY_ETH_PRICE)
+
+      const gasCostInNativeCurrency = formatEther(gasPrice.mul(GAS_COST))
+
+      gas = `${(Number(gasCostInNativeCurrency) * Number(nativeCurrencyPrice)).toFixed(2)}$`
+    } catch (e) {
+      gas = undefined
+    }
+
+    const details = {
+      fee,
+      gas,
+      receiveAmount: Number(formatUnits(toAmount.toString(), decimals)).toFixed(2),
+      estimateTime: '5 min',
+      requestId: helperRequestId,
+    }
+    this.store.dispatch(this.actions.setBridgeDetails(details))
+  }
+
+  private _fetchHistory = async () => {
+    try {
+      if (!this._account || !this._activeChainId) return
+
+      const [{ requests: homeRequests }, { requests: foreignRequests }] = await Promise.all<
+        SubgraphRequestsData,
+        SubgraphRequestsData
+      >([
```

Comment:
Single type passed as arg will be assumend for all elements in arr
```suggestion
      const [{ requests: homeRequests }, { requests: foreignRequests }] = await Promise.all<
        SubgraphRequestsData
      >([
```

Reply:
fixed [a2c2761](https://github.com/levelkdev/swapr-dapp/pull/1007/commits/a2c2761ff4b65a4f7e3d129ecd0d7e484177d0b0)